runtime/compiler/optimizer/EscapeAnalysis.cpp

```diff
@@ -2332,6 +2432,32 @@ bool TR_EscapeAnalysis::checkOverlappingLoopAllocation(TR::Node *node, TR::Node
    }
 
 
+
+void TR_EscapeAnalysis::collectAliasesOfAllocations(TR::Node *node, TR::Node *allocNode, bool collectAliases)
```

Comment:
I am not a fan of this bool parameter that makes this method just do the visit - basically putting all of the logic in an if... I would rather we had a method to just visit and a method to collect aliases and dropped this boo parameter which is going to make usage harder to understand/follow.

Reply:
Andrew @andrewcraik, Just to be sure I'm clear on what you're suggesting, do you want to see a `visitTree` method that just visits nodes in a tree and a `collectAliasesOfAllocations` that both visits and collects aliases unconditionally?  Or do you want to see a `visitTree` method that visits nodes and also takes a function pointer (or an object) that it calls at each node, where that function would either be no-op or collect aliases?

Reply:
for this I'd suggest the former - a `visitTree` that unconditionally visits the tree and marks it as visited and a `collectAliasesofAllocations` that does what it currently does when the Boolean is true. Function pointers can make debugging the compiler harder so I try to avoid them.

Reply:
Split `collectAliasesOfAllocations` into two methods in commit 665010adf18731078ee8181b05cf4ae0480c1a69, as Andrew suggested